### PR TITLE
feat: rewrite route-with-jwt-plugin-test using ginkgo

### PR DESCRIPTION
**Pull request by bisakhmondal**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

**Related issues**

#1500 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: api/test/e2enew/route/route_with_plugin_jwt_test.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+			Path:   "/apisix/admin/routes/r1",
+			Body: `{
+				 "name": "route1",
+				 "uri": "/hello",
+				 "plugins": {
+					 "jwt-auth": {}
+				 },
+				 "upstream": {
+					 "type": "roundrobin",
+					"nodes": {
```
Comment (starsz) [inline api/test/e2enew/route/route_with_plugin_jwt_test.go:57]: Code style.

File: api/test/e2enew/route/route_with_plugin_jwt_test.go [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+			ExpectBody:   []string{`"code":0`, `"id":"r1"`, `"uri":"/hello"`, `"name":"route1"`, `"jwt-auth":{}`},
+			ExpectStatus: http.StatusOK,
+		}),
+	)
+
+	ginkgo.It("get the jwt token", func() {
+		time.Sleep(base.SleepTime)
+
+		request, _ := http.NewRequest("GET", base.APISIXHost+"/apisix/plugin/jwt/sign?key=user-key", nil)
+		request.Header.Add("Authorization", base.GetToken())
```
Comment (starsz) [inline api/test/e2enew/route/route_with_plugin_jwt_test.go:240]: Why do we need to add an Authorization head here?

Comment (bisakhmondal) [reply api/test/e2enew/route/route_with_plugin_jwt_test.go:240]: Ohh, we don't. It's the APISIXHost. Let me check. It was in the e2e test suite code.
Btw good catch :+1:

File: api/test/e2enew/route/route_with_plugin_jwt_test.go (hunk 1 above)
Comment (bisakhmondal) [reply api/test/e2enew/route/route_with_plugin_jwt_test.go:57]: fixed

**Review by nic-chen — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**